src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosDriverFactory.scala

```diff
@@ -1,59 +1,47 @@
 package org.apache.mesos.chronos.scheduler.mesos
 
-import java.io.{ FileInputStream, IOException }
-import java.nio.file.attribute.PosixFilePermission
-import java.nio.file.{ Files, Paths }
 import java.util.logging.Logger
 
-import com.google.protobuf.ByteString
-import org.apache.mesos.Protos.{ Credential, FrameworkInfo, Status }
+import mesosphere.chaos.http.HttpConf
+import mesosphere.mesos.util.FrameworkIdUtil
+import org.apache.mesos.Protos.{ FrameworkID, Status }
 import org.apache.mesos.chronos.scheduler.config.SchedulerConfiguration
-import org.apache.mesos.{ MesosSchedulerDriver, Scheduler, SchedulerDriver }
-
-import scala.collection.JavaConverters.asScalaSetConverter
+import org.apache.mesos.{ Scheduler, SchedulerDriver }
 
 /**
  * The chronos driver doesn't allow calling the start() method after stop() has been called, thus we need a factory to
  * create a new driver once we call stop() - which will be called if the leader abdicates or is no longer a leader.
  * @author Florian Leibert (flo@leibert.de)
  */
-class MesosDriverFactory(val mesosScheduler: Scheduler, val frameworkInfo: FrameworkInfo, val config: SchedulerConfiguration) {
+class MesosDriverFactory(
+                          val scheduler: Scheduler,
+                          val frameworkIdUtil: FrameworkIdUtil,
```

Comment:
why is this a `val`? We should avoid exporting dependencies as fields whenever possible.